---
title: 'Jointly Modelling SNPs<br>with<br>Survival & Longitudinal Trait?'
author: 'MickaÃ«l CANOUIL, *Ph.D.*'
date: 'Friday, 18<sup>th</sup> of October (2018)'
monofont: 'Source Code Pro'
monofontoptions: 'Scale=0.7'
bibliography: [bib/Canouil_etal_2018.bib]
biblio-style: apalike
csl: csl/apa.csl
params:
  eval: TRUE
output:
  ioslides_presentation:
    css: 'template/shiny-slides.css'
    logo: 'template/logo_UMR.png'
    smaller: false
    self_contained: true
    mathjax: default
    transition: 0.5
    incremental: false
---


```{r setup, include = FALSE}
options(stringsAsFactors = FALSE)
# Sys.setlocale("LC_TIME", "english_united kingdom.1252")

output_directory <- ""

### Load packages and functions
library(tidyverse)
library(broom)
library(scales)
library(parallel)
library(grid)
library(knitr)
library(rmarkdown)
library(kableExtra)
library(gganimate)

devtools::source_url('https://github.com/mcanouil/DEV/raw/master/Rfunctions/theme_black.R')
devtools::source_url('https://github.com/mcanouil/DEV/raw/master/Rfunctions/pretty_kable.R')

options("width" = 80)
### Set knitr rmarkdown chunk options
opts_chunk$set(
  include = TRUE,
  echo = FALSE,
  warning = FALSE,
  message = FALSE,
  eval = params$eval,
  tidy = FALSE,
  crop = TRUE,
  autodep = TRUE,
  dpi = 120,
  fig.path = "./images/",
  cache = FALSE,
  # cache.path = NULL,
  width = 80,
  comment = "#>",
  results = "asis",
  fig.height = floor(10/3 * 10)/10* 0.9, 
  fig.width = floor(16/9 * 10/3 *10)/10 * 0.9
)

### Define theme
theme_set(theme_black(base_size = 14))

# options(tibble.print_max = 3, tibble.print_min = 3)

require(ggraph)
jm_data <- new.env()
load(file = "Routput/DataArticle_20180102.Rdata", envir = jm_data)
format_scientific <- function(x, digits = 3) {
  y <- format(x, scientific = -1, digits = digits)
  if (length(grep("e", y)) != 0) {
    return(
      paste0(
        gsub("e.*", "", format(x, scientific = -1, digits = digits)), 
        "\\times 10^{", 
        gsub(".*e", "", format(x, scientific = -1, digits = digits)), "}"
      )
    )
  } else {
    return(y)
  }
}

```



## T2D and Associated Traits {.flexbox .vcenter}
```{r, out.height = "405px", out.width = "720px"}
include_graphics(path = "images/ProkopenkoMetaboDiagram.png")
```
<p>@marullo_insights_2014</p>


## A lot of SNPs discovered, but ... {.flexbox .vcenter}
```{r, out.height = "405px", out.width = "720px"}
include_graphics(path = "images/T2Dhistory.jpg")
```
<p>@flannick_type_2016</p>


# Glycaemia and Type 2 Diabetes | A Genetic Joint Effect? {.flexbox .vcenter}

<p class="auto-fadein" align="center" style="line-height: 175%;">
_"In all cases, the glucose-raising allele was associated with increased risk of T2D, yet fasting glucose effect sizes and T2D ORs were weakly correlated"_
</p>
<p class="auto-fadein">
@scott_large-scale_2012
</p>


## Glycaemia and Type 2 Diabetes {.flexbox .vcenter}
```{r, out.height = "405px", out.width = "720px"}
include_graphics(path = "images/ng2385-F2.png")
```
<p>@scott_large-scale_2012</p>

## Glycaemia and Type 2 Diabetes {.flexbox .vcenter}
```{r, out.height = "405px", out.width = "720px"}
include_graphics(path = "images/Yaghootkar.png")
```
<p>@yaghootkar_recent_2013</p>


# Joint Model | What for?<br>&<br>What is it? {.flexbox .vcenter}

## What for? {.flexbox .vcenter}

* To identify biomarker relavent to a disease

* To identify the effect of a treatment on a disease  
    (independently of the association between the disease and the biomarker)

*For example:*

  * __Biomarker__: fasting glucose
  * __Event__: Type 2 Diabetes


## Let's start with the Cox Model...

(Extended) Cox model:
$$\begin{align}
\lambda_i(t)=\lambda_0(t) \exp(\beta Y_i(t) + \alpha Z_i + \eta W_i)
\end{align}$$

where:

* $\lambda_i(t)$ is the hazard function at time $t$ for individual $i$;
* $\lambda_0(t)$ is the unspecified baseline hazard function;
* $\alpha$ measures the effect of $Z_i$ on the hazard function;
* $\beta$ measures the association between the trajectory function $Y_i(t)$ and the hazard function.


## It works, but biomarkers...

1. are measured at determined time points ($t_{ij}$)
2. can have missing values over time
    * => Imputation? 
    * $\Rightarrow$ Bias introduction
3. are measured with some degree of error
    * $\Rightarrow$ Noise in the biomarker trajectory ($Y_i(t_{ij}) \neq X_i(t_{ij})$)
4. can be endogenous
    * $\Rightarrow$ Trajectory can change when the event occurs
    * $\Rightarrow$ Bias introduction


## Hopefully, the Mixed Model is there!

(Generalised) linear mixed effect (LME) model:
$$Y_{i}(t_{ij})=X_{i}(t_{ij})+\epsilon_{i}(t_{ij})$$

where:

* $Y_{i}(t_{ij})$ is the observed value 
* $X_{i}(t_{ij})$ is the true (unobserved) value of the longitudinal measurement at time $t_{ij}$ for individual $i$. 
* $\epsilon_{i}(t_{ij})$ is a random error term, usually:  
    $$\epsilon_{i}(t_{ij})\sim \mathcal{N}(0,\sigma^2)$$


## What can we do with a Joint Model?

Test simultaneously an effect on:

* a biomarker ($\gamma$);
* an event ($\alpha$);
* a biomarker and an event ($\beta\gamma+\alpha$).


## The best part?

A gain in statistical power to detect those effects

* if $\beta\neq0$  (to detect a joint effect of $Z$: $\beta\gamma+\alpha\neq0$) @chen_sample_2011;
* compared to the extended Cox model.


## What is Joint Model? With a picture! {.flexbox .vcenter}

```{r}
data_arrows <- tribble(
  ~x, ~y, ~xend, ~yend, ~step,
  
  0.15, 1, 0.85, 1, 2,
  0.20, 1, 0.80, 1, 3,
  0.20, 1, 0.80, 1, 4,
  0.20, 1, 0.80, 1, 5,
  0.20, 1, 0.80, 1, 6,
  0.20, 1, 0.80, 1, 7,
  0.20, 1, 0.80, 1, 8,
  
  1.20, 1, 1.85, 0.55, 4,
  1.20, 1, 1.85, 0.55, 8,
  
  1.15, 0, 1.85, 0.45, 5,
  1.15, 0, 1.85, 0.45, 7,
  1.15, 0, 1.85, 0.45, 8,
  
  1, 0.1, 1, 0.9, 6,
  1, 0.1, 1, 0.9, 7,
  1, 0.1, 1, 0.9, 8
)

data_labels <- tribble(
  ~x, ~y, ~label, ~colour, ~step,
  1, 1, "Y(t)", 1, 1,
  2, 0.5, "S", 1, 1,
  1, 0, "Z", 1, 1,
  
  0, 1, "Y(t)", 1, 2,
  1, 1, "X(t)", 1, 2,
  2, 0.5, "T2D", 1, 2,
  1, 0, "SNP", 1, 2,
  0.5, 1.1, "epsilon", 2, 2,
  
  0, 1, "FG[obs]", 1, 3,
  1, 1, "FG[true]", 1, 3,
  2, 0.5, "T2D", 1, 3,
  1, 0, "SNP", 1, 3,
  0.5, 1.1, "epsilon", 2, 3,
  
  0, 1, "FG[obs]", 1, 4,
  1, 1, "FG[true]", 1, 4,
  2, 0.5, "T2D", 1, 4,
  1, 0, "SNP", 1, 4,
  0.5, 1.1, "epsilon", 2, 4,
  1.5, 0.9, "beta", 2, 4,
  
  0, 1, "FG[obs]", 1, 5,
  1, 1, "FG[true]", 1, 5,
  2, 0.5, "T2D", 1, 5,
  1, 0, "SNP", 1, 5,
  0.5, 1.1, "epsilon", 2, 5,
  1.5, 0.10, "alpha", 2, 5,
  
  0, 1, "FG[obs]", 1, 6,
  1, 1, "FG[true]", 1, 6,
  2, 0.5, "T2D", 1, 6,
  1, 0, "SNP", 1, 6,
  0.5, 1.1, "epsilon", 2, 6,
  0.9, 0.5, "gamma", 2, 6,
  
  0, 1, "FG[obs]", 1, 7,
  1, 1, "FG[true]", 1, 7,
  2, 0.5, "T2D", 1, 7,
  1, 0, "SNP", 1, 7,
  0.5, 1.1, "epsilon", 2, 7,
  0.9, 0.5, "gamma", 2, 7,
  1.5, 0.10, "alpha", 2, 7,
  
  0, 1, "FG[obs]", 1, 8,
  1, 1, "FG[true]", 1, 8,
  2, 0.5, "T2D", 1, 8,
  1, 0, "SNP", 1, 8,
  0.5, 1.1, "epsilon", 2, 8,
  1.5, 0.10, "alpha", 2, 8,
  0.9, 0.5, "gamma", 2, 8,
  1.5, 0.9, "beta", 2, 8
)
p_init <- ggplot() +
  theme(
    panel.grid = element_blank(), 
    axis.title = element_blank(), 
    axis.text = element_blank(), 
    axis.ticks = element_blank(), 
    panel.border = element_blank(),
    legend.position = "none"
  )
  p <- p_init +
    geom_segment(
      data = data_arrows %>% filter(step==1),
      mapping = aes(x = x, xend = xend, y = y,  yend = yend),
      colour = "white",
      arrow = arrow(length = unit(8, "point"), type = "closed"),
      lineend = "round",
      linejoin = "round"
    ) +
    geom_text(
      data = data_labels %>% filter(step==1), 
      mapping = aes(x = x, y = y, label = label, colour = factor(colour)), 
      size = 6,
      parse = TRUE
    ) +
    scale_x_continuous(expand = expand_scale(mult = 0.2)) +
    scale_y_continuous(expand = expand_scale(mult = 0.2)) +
    scale_colour_viridis_d() +
    coord_cartesian(xlim = c(0, 2), ylim = c(0, 1))
  print(p)
cat("\n")
```

```{r}
for (istep in 2:8) {
  cat("\n## What is Joint Model? With a picture! {.flexbox .vcenter}\n")
  cat("\n")
  p <- p_init +
    geom_segment(
      data = data_arrows %>% filter(step==istep),
      mapping = aes(x = x, xend = xend, y = y,  yend = yend),
      colour = "white",
      arrow = arrow(length = unit(8, "point"), type = "closed"),
      lineend = "round",
      linejoin = "round"
    ) +
    geom_text(
      data = data_labels %>% filter(step==istep), 
      mapping = aes(x = x, y = y, label = label, colour = factor(colour)), 
      size = 6,
      parse = TRUE
    ) +
    scale_x_continuous(expand = expand_scale(mult = 0.2)) +
    scale_y_continuous(expand = expand_scale(mult = 0.2)) +
    scale_colour_viridis_d() +
    coord_cartesian(xlim = c(0, 2), ylim = c(0, 1))
  print(p)
  cat("\n")
}

# p <- p +
#   transition_states(
#     states = step,
#     transition_length = 0,
#     state_length = 2,
#     wrap = FALSE
#   )
# animate(
#   plot = p, 
#   nframes = 400,
#   width = 800, 
#   height = 450,
#   res = 120,
#   bg = ggplot2::theme_get()$plot.background$colour,
#   renderer = ffmpeg_renderer(options = list(pix_fmt = "yuv420p", "r" = 0.9))
# )

```

## What is Joint Model? With equations!

The standard (joint likelihood) formulation involves two components:

* a longitudinal component
* a time-to-event (survival) component.

Let:

* $n$, the sample size;  
* $i$, an individual ($i=1,\cdots,n$);
* $m_i$, the number of measurements on individual $i$;
* $t_{ij}$, a time points ($j=1,\cdots,m_i$).


## The longitudinal component

(Generalised) linear mixed effect (LME) model:
$$Y_{i}(t_{ij})=X_{i}(t_{ij})+\epsilon_{i}(t_{ij})$$
$X_{ij}$ is the trajectory function, and could be defined:
$$\begin{gather}X_{i}(t_{ij})=\theta_{0i} + \theta_{1i}t_{ij} + \cdots + \theta_{pi}t_{ij}^p &, & \boldsymbol\theta_p \sim \mathcal{N}(\boldsymbol\mu_, \boldsymbol\Sigma)\end{gather}$$

For simplicity here, we assume linearity over time ($\theta_{0i}+\theta_{1i}t_{ij}$):
$$\begin{gather}Y_{i}(t_{ij})=\theta_{0i}+\theta_{1i}t_{ij}+\gamma Z_i+\delta W_i+\epsilon_{ij} &, & \boldsymbol{\theta} \sim \mathcal{N}_2 (\boldsymbol{\mu},\boldsymbol{\Sigma})\end{gather}$$

## The longitudinal component

(Generalised) linear mixed effect (LME) model:
$$Y_{i}(t_{ij})=\theta_{0i}+\theta_{1i}t_{ij}+\gamma Z_i+\delta W_i+\epsilon_{ij}$$

where:

* $Y_{i}(t_{ij})$, the observed value;
* $X_{i}(t_{ij})$, the true (unobserved) value of the longitudinal measurement at time $t_{ij}$ for individual $i$;
* $\epsilon_{ij}$ is a random error term, usually: $\epsilon_{ij}\sim \mathcal{N}(0,\sigma^2)$;
* $Z_i$, a vector denoting the genotype of individual $i$;
* $W_i$, a set of adjusting covariates.


## The time-to-event component

(Extended) Cox model (proportional hazards):
$$\begin{align}
\lambda_i(t)&=\lim_{dt \to 0} \frac{P\{t\leq T_i<t+dt|T_i\geq t, \bar{Y_i}(t), Z_i, W_i\}}{dt}\\
&=\lambda_0(t) \exp\{\beta X_{i}(t) + \alpha Z_i + \eta W_i\}
\end{align}$$

Where:

* $\bar{Y_i}(t)=\{Y_i(u),0 \leq u \leq t\}$, the history of the trajectory;
* $T_i$,  the event time for individual $i$;
* $C_i$, the right censoring time (end of the follow-up);
* $\Delta_i$, an event indicator: $\begin{cases}
    \Delta_i=0, & \text{if }\ T_i>C_i. \\
    \Delta_i=1, & \text{if }\ T_i <= C_i.
    \end{cases}$


## Hypothesis testing

Null hypothesis: $\begin{cases}H_0&:& \theta=\theta_0\\H_1&:& \theta\neq\theta_0\end{cases}$

* __Likelihood Ratio Test__  
    $LRT=-2\{\ell(\hat{\theta}_0)-\ell(\hat{\theta})\}$

*  __Wald Test__  
    $W=(\hat{\theta}-\theta_0)^\top \mathcal{I}(\hat{\theta})(\hat{\theta}-\theta_0)$  
    (univariate: $(\hat{\theta}_j-\theta_{0j})/\widehat{s.e.}(\hat{\theta}_j)$)

* __Score Test__  
  $U=S^\top(\hat{\theta}_0)\{\mathcal{I}(\hat{\theta}_0)\}^{-1}S(\hat{\theta}_0)$


# Is the Joint Model worth it? | Let's find out with simulations!

## Estimators (& Computation time)

* Is Joint Model estimators good?  
    Bias, variance and RMSE (Root-Mean Square Error)  
    $$\begin{align}
    \operatorname{MSE}(\hat\phi)&= \operatorname{Bias}(\hat\phi)^2 + \operatorname{Var}(\hat\phi)\\
    \operatorname{RMSE}(\hat{\phi})&=\sqrt{\operatorname{MSE}(\hat\phi)}\\
    &=\sqrt{E\{(\hat{\phi}-\phi)^2\}}
    \end{align}$$
    $$\phi=(\beta, \gamma, \alpha)$$

* Can we do a whole genome analysis...  
    ... in a reasonable time frame?


## Let's find a more "naive" approach!

What if, we split the job in two?

$\Rightarrow$ "Two-Step" [@tsiatis_modeling_1995]?

1. (Generalised) linear mixed effect (LME) model  
    $$\begin{align}
    Y_{i}(t)&=X_i(t)+ \epsilon_{i}(t)\\
    X^*_{i}(t)&=E\{X_{i}(t)|\bar{Y_i}(t), T_i\geq t\}
    \end{align}$$

2. (Extended) Cox model (proportional hazards)  
    $$h_i(t)=h_0(t) \exp\{\beta X^*_{i}(t)\}$$


## Time to generate fake data!

Let's keep it simple, i.e., without covariates:

* the trajectory: $Y_{i}(t)=\theta_{0i} + \theta_{1i}t + \gamma Z_i + \epsilon_{i}(t)$

* the event: $\lambda_i(t)=\lambda_0(t) \exp\{\beta X_{i}(t) + \alpha Z_i\}$

* the time of event, following the (classical) exponential distribution [@austin_generating_2012]:  

    $$\begin{gather}
    H_i(T_i)=\int_0^{T_i}\lambda_0(t) \exp(\beta X_i(t)+\alpha Z_i)dt , & \lambda_0(t)=\lambda\\
    F_i(T_i)=1-exp(-H_i(T_i))=u , & u\sim\mathcal{U}(0, 1)
    \end{gather}$$
    $$T_i=\frac{1}{\beta\theta_{1i}}\log\left(1-\frac{\beta\theta_{1i}\times \log(1-u)}{\lambda \exp(\beta\theta_{0i}+(\beta\gamma+\alpha)Z_i)}\right)$$

## Set the parameters! {.flexbox .vcenter}

<div class="columns-2">
```{r, out.height = "360px", out.width = "360px"}
include_graphics(path = "images/ng2385-F2.png")
```
<p>@scott_large-scale_2012</p>

```{r, out.height = "360px", out.width = "360px"}
include_graphics(path = "images/Yaghootkar.png")
```
<p>@yaghootkar_recent_2013</p>
</div>


## Set the parameters! {.flexbox .vcenter}
```{r}
settings.table <- data.frame(
  matrix(NA, nrow = 9, ncol = 2, dimnames = list(NULL, c("Parameters", "Values")))
)
settings.table[, "Parameters"] <- c(
  "Number of participants ($n$)", 
  "Number of measures ($m$)", 
  "Diabetes incidence rate ($d$)", 
  "Minor allele frequency ($f$)", 
  "Random effects ($\\theta$)", 
  "SNP effect on $Y_{ij}$ ($\\gamma$)", 
  "SNP effect on $T_i$ ($\\alpha$)", 
  "Association between $Y_{ij}$ and $T_i$ ($\\beta$)", 
  "Error term ($\\epsilon$)"
)
settings.table[, "Values"] <- c(
  format(jm_data$rawsettings$n+1, big.mark = ","), 
  4, 
  signif(jm_data$rawsettings$d, digits = 3), 
  signif(jm_data$rawsettings$maf, digits = 3),
  paste0(
    "$\\sim\\mathcal{N}_2\\left (\\begin{bmatrix}", 
    format_scientific(jm_data$rawsettings$thetas[1]), "\\\\", 
    format_scientific(jm_data$rawsettings$thetas[2]), 
    "\\end{bmatrix} , \\begin{bmatrix} ", 
    format_scientific(jm_data$rawsettings$Epsilon_thetas[1, 1]), " & ", 
    format_scientific(jm_data$rawsettings$Epsilon_thetas[1, 2]), " \\\\ ", 
    format_scientific(jm_data$rawsettings$Epsilon_thetas[2, 1]), " & ", 
    format_scientific(jm_data$rawsettings$Epsilon_thetas[2, 2]), " \\end{bmatrix} \\right )$"
  ),
  signif(jm_data$rawsettings$gamma, digits = 3), 
  signif(jm_data$rawsettings$alpha[2], digits = 3), 
  signif(jm_data$rawsettings$beta, digits = 3), 
  paste0("$\\sim\\mathcal{N}(0,", signif(jm_data$rawsettings$sigma, digits = 3), "^2)$")
)


kable(
  x = settings.table, 
  format.args = list(scientific = -1, digits = 3, big.mark = ","),
  align = c("l", "c"),
  escape = FALSE,
  caption = paste0(
    'Parameters and numerical values used for sensitivity analysis and simulations, based on results from ',
    jm_data$bestEffect[, "snp"],
    ' within gene _',
    jm_data$bestEffect[, "closestgene"],
    '_ in the French cohort D.E.S.I.R.'
  ),
  table.attr = 'style="font-size:75%;"'
)
```


# <img src="http://mickael.canouil.fr/img/IDPhotoBWlight.png" height = "150px"></img>__*Me?*__ {.flexbox .vcenter}

<div class="auto-fadein">
<script src="https://cdnjs.cloudflare.com/ajax/libs/uikit/3.0.0-rc.10/js/uikit.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/uikit/3.0.0-rc.10/js/uikit-icons.min.js"></script>
<div class="columns-2">

<span class="uk-icon-button" uk-icon="icon: receiver; ratio: 1"></span>
    <a style="border-bottom: none; font-size: 80%; vertical-align: text-top;" href="" target="_blank">+33 (0) 374 00 81 29</a> 
    
<span class="uk-icon-button" uk-icon="icon: mail; ratio: 1"></span>
    <a style="border-bottom: none; font-size: 80%; vertical-align: text-top;" href="mailto:mickael.canouil@cnrs.fr" target="_blank">mickael.canouil@cnrs.fr</a> 

<span class="uk-icon-button" uk-icon="icon: home; ratio: 1"></span>
    <a style="border-bottom: none; font-size: 80%; vertical-align: text-top;" href="http://mickael.canouil.fr" target="_blank">mickael.canouil.fr</a> 

<span class="uk-icon-button" uk-icon="icon: linkedin; ratio: 1"></span>
    <a style="border-bottom: none; font-size: 80%; vertical-align: text-top;" href="https://www.linkedin.com/in/mickael-canouil" target="_blank">mickael-canouil</a> 

<span class="uk-icon-button" uk-icon="icon: github; ratio: 1"></span>
    <a style="border-bottom: none; font-size: 80%; vertical-align: text-top;" href="https://github.com/mcanouil" target="_blank">mcanouil</a> 

<span class="uk-icon-button" uk-icon="icon: twitter; ratio: 1"></span>
    <a style="border-bottom: none; font-size: 80%; vertical-align: text-top;" href="https://twitter.com/Coeos_" target="_blank">@ Coeos_</a> 

</div>
</div>


# References {.smaller}
